File: Data Structures & Algorithms/append-characters-to-string-to-make-subsequence/submission-2.py
class Solution:
    def appendCharacters(self, s: str, t: str) -> int:

        sp = 0
        tp = 0
        while tp < len(t) and sp < len(s):
            if s[sp] == t[tp]:
                tp += 1
            sp += 1
        return len(t) - tp
        

File: Data Structures & Algorithms/maximum-difference-between-even-and-odd-frequency-i/submission-2.java
class Solution {
    public int maxDifference(String s) {
        HashMap<Character, Integer> map = new HashMap();
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            map.merge(c, 1, (oldV, newV) -> oldV + 1);
        }

        int maxOdd = 0;
        int minEven = Integer.MAX_VALUE;

        for (int value : map.values()) {
            System.out.println(value);
            if (value % 2 == 0 && value < minEven) {
                minEven = value;
            }
            else if (value % 2 != 0 && value > maxOdd) {
                maxOdd = value;
            }
        }

        return maxOdd - minEven;
    }
}